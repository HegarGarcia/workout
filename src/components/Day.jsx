import Typography from '@material-ui/core/Typography';
import React, { memo } from 'react';
import styled, { css } from 'styled-components';

const DayContainer = styled.button`
  display: flex;
  flex-direction: column;
  background: transparent;
  border: none;
  color: white;
  outline: none;
  text-transform: uppercase;
  text-align: center;
  align-items: center;

  p {
    color: rgba(255, 255, 255, 0.36);
  }
`;

const StyledDay = styled.div`
  height: 30px;
  width: 30px;
  border-radius: 50%;

  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.25s ease-in-out;

  ${({ active }) => {
    if (!active) {
      return css`
        background: transparent;
      `;
    }

    return css`
      background: #768fff;
    `;
  }}}
    
  span {
    font-size: 11px;
    font-weight: 500;
    margin-left: 1px;
  }
`;
function Day({ date, active }) {
  return (
    <DayContainer data-date={date.date}>
      <Typography variant="overline" component="p">
        {date.weekDay}
      </Typography>
      <StyledDay active={active}>
        <span>{date.number}</span>
      </StyledDay>
    </DayContainer>
  );
}

export default memo(Day);
