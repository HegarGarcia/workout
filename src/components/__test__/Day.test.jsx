import React from 'react';

import { render, screen, cleanup } from '@testing-library/react';
import Day from '../Day';

describe('<Day /> spec', () => {
  const date = {
    date: '2020 06 17',
    weekDay: 'WED',
    number: 17
  };

  afterEach(cleanup);

  it('Should not regress', () => {
    const { container } = render(<Day date={date} />);
    expect(container).toMatchSnapshot();
  });

  it('Should show the day number', () => {
    render(<Day date={date} />);
    expect(screen.getByText(date.number.toString())).toBeTruthy();
  });

  it('Should show the week day nem', () => {
    render(<Day date={date} />);
    expect(screen.getByText(date.weekDay)).toBeTruthy();
  });

  it('Should save date in dataset', () => {
    render(<Day date={date} />);
    expect(screen.getByTestId('day-container').dataset.date).toBe(date.date);
  });
});
