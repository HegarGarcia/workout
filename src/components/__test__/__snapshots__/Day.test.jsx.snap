// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> spec Should not regress 1`] = `
<div>
  <button
    class="sc-AxjAm gnohHb"
    data-date="2020 06 17"
    data-testid="day-container"
  >
    <p
      class="MuiTypography-root MuiTypography-overline"
    >
      WED
    </p>
    <div
      class="sc-AxirZ lnryAL"
    >
      <span>
        17
      </span>
    </div>
  </button>
</div>
`;
